synapse/lib/ast.py

```diff
@@ -2816,6 +2853,174 @@ async def run(self, runt, genr):
 
             await asyncio.sleep(0)
 
+class N1Walk(Oper):
```

Comment:
N1Walk and N2Walk are could use a refactoring to reduce the amount of duplicate code in them.